### ISSUE TITLE
Sanitize.clean has issues with malformed/missing </script> closing tags

### ISSUE THREAD

**Issue by jrjamespdx** [rgrove/sanitize#142]
When using Sanitize.clean and passing in the following line, everything behaves as expected:

Input:  `<html><b>Test</b><script src="file.js"></script></html>`
Output:  `"Test"`

However in the following cases, the behavior is a bit odd:

Input: `<html><b>Test</b><script src="file.js"></scrip></html>`
Output:  `"Test&lt;/scrip&gt;&lt;/html&gt;"`

Input: `<html><b>Test</b><script src="file.js"></html>`
Output:  `"Test&lt;/html&gt;"`

Even when using the `:remove_contents => true` option, the behavior is off:

Input:  `'<html>Test<script src="file.js"></scrip>Bob</html>Steve'`
Output:  `"Test"`

Input: `<html>Test<script src="file.js"></script>Bob</html>Steve`
Output: `"TestBobSteve"`

Input: `<html>Test<script src="file.js">Bob</html>Steve`
Output: `"Test"`

Input: `<html>Test<script src="file.js">Allan</script>Bob</html>Steve`
Output: `"TestBobSteve"`

Input: `<html>Test<script src="file.js">Allan</scrip>Bob</html>Steve`
Output: `"Test"`

Input: `<html>Test<script src="file.js">Allan Bob</html>Steve`
Output: `"Test"`

**Comment by rgrove** [comment]
tl;dr: Sanitize is behaving correctly here, even though it's not entirely intuitive.

The key thing to understand is that when there's no closing `</script>` tag, the script element consumes everything that follows it until the end of the input, and that becomes the content of a text node inside the script element. So given the input `<html>foo<script></scrip>bar</html>`, the parsed DOM tree looks like this (minus auto-inserted elements like `<head>` and `<body>`, which I'm omitting for clarity):

-   Element: `<html>`
    -   Text: `foo`
    -   Element: `<script>`
        -   Text: `</scrip>bar</html>`

When `remove_contents` is `false`, Sanitize removes the script element and hoists its text content up, escaping it appropriately when the tree is serialized. Thus you get `foo&lt;/scrip&gt;bar&lt;/html&gt;`.

When `remove_contents` is `true`, Sanitize removes the script element and its text content, thus you get only `foo` (because everything after the opening `<script>` tag becomes text content of the script element).

Hope this helps!
